guiEditor/src/diagram/workbench.tsx

```diff
@@ -464,7 +503,17 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         const onPointerDown = guiControl.onPointerDownObservable.add((evt) => {
             if (!this.isUp || evt.buttonIndex > 0) return;
             if (this._forceSelecting) {
-                this.isSelected(true, guiControl);
+                // if this is our first click or the clicked control is a parent of the previous clicked control.
+                if (!this._doubleClick && this._isMainSelectionParent(guiControl)) {
+                    this._doubleClick = guiControl;
+                    window.setTimeout(() => {
+                        this._doubleClick = null;
+                    }, DOUBLE_CLICK);
+                }
+                else { //select our new main control.
+                    this.determineMouseSelection(guiControl);

```

Comment:
should the second click control be checked against _doubleClick control to validate the action ? (wondering what happens if double click but not on the same position)

Reply:
If it is a different control, it will catch it in the next function. 
determineMouseSelection will either go down the tree or restart tree at new main control. Ill update the comment to reflect this,